newsroom/mgmt_api/companies_products.py

```diff
@@ -0,0 +1,107 @@
+    return company.products or []
+
+
+company_products_endpoints = EndpointGroup("company_products", __name__, url_prefix="/api")
+
+
+class CPCompanyProduct(CompanyProduct):
+    # Declare section as optional
+    section: Optional[SectionEnum] = None  # type: ignore
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        """Allow `product` key as an alias for `_id`"""
+        if "product" in data:
+            data["_id"] = data.pop("product")
+        return super().from_dict(data)
+
+
+class CompanyProductRouteArguments(BaseModel):
+    company_id: fields.ObjectId
+
+    async def get_company(self) -> CompanyResource | None:
+        return await CompanyResource.get_service().find_by_id(self.company_id)
+
+
+@company_products_endpoints.endpoint("/api/companies/<regex('[a-f0-9]{24}'):company_id>/products", methods=["POST"])
+async def update_company_products(args: CompanyProductRouteArguments, params: None, request: Request) -> Response:
+    company = await args.get_company()
+    if not company:
+        return Response({"error": "Company not found"}, status=404)
+
+    product_links_json = await request.get_json()
+    product_links: list[CPCompanyProduct] = [CPCompanyProduct.from_dict(link) for link in product_links_json]
+    ids: List = []
+
+    company_products = get_company_products(company)
+    updated_products = company_products[:]
+
+    for doc in product_links:
+        product_id = doc._id
+        if not product_id:
+            continue
+
+        product_data = await ProductsService().find_by_id(ObjectId(product_id))
+        if not product_data:
+            continue
+
+        updated_products = [p for p in updated_products if p._id != product_data.id]
+
+        if doc.link:
+            updated_products.append(get_product_ref(product_data, doc.seats).to_dict())
+            pass
+
+        ids.append(product_id)
+
+    if ids:
+        await CPCompaniesService().system_update(ObjectId(company.id), {"products": updated_products})
+
+    return Response({"updated_product_ids": ids}, 201)
+
+
+@company_products_endpoints.endpoint("/api/companies/<regex('[a-f0-9]{24}'):company_id>/products", methods=["GET"])
+async def get_company_products_endpoint(args: CompanyProductRouteArguments, params: None, request: Request) -> Response:
+    company = await args.get_company()
+    if not company:
+        return Response({"error": "Company not found"}, status=404)
+
+    company_products = get_company_products(company)
+
+    products_data = []
+    for product in company_products:
+        product_data = await ProductsService().find_by_id(ObjectId(product._id))
+        if product_data:
+            item = {
+                "name": product_data.name,
+                "description": product_data.description,
+                "query": product_data.query,
+                "product_type": product_data.product_type,
+                "seats": product.seats or 0,
```

Comment:
The `seats` attribute should be coming from the company, like in the [original implementation](https://github.com/superdesk/newsroom-app-cp/blob/cn-develop/server/cp/mgmt_api/companies.py#L136).

Reply:
that `product` is coming from the `company_product`  see  [here](https://github.com/superdesk/newsroom-core/pull/1237/files/c4cbf76a5ca9dc0d68bde8b689f082e25042a9ec#diff-a21486d9ea0628c77acc3acd5c5a9b7e1130938f23d53ed3a8e222f7dd321dbbR85) and [here](https://github.com/superdesk/newsroom-core/pull/1237/files/c4cbf76a5ca9dc0d68bde8b689f082e25042a9ec#diff-a21486d9ea0628c77acc3acd5c5a9b7e1130938f23d53ed3a8e222f7dd321dbbR88)

Reply:
Oh, whoops I misread the value name